exercises/simple-cipher/canonical-data.json

```diff
@@ -0,0 +1,145 @@
+{
+  "exercise": "simple-cipher",
+  "version": "1.0.0",
+  "cases": [
+    {
+      "description": "Random key cipher",
```

Comment:
Reading the [exercise's description](https://github.com/gustavosobral/problem-specifications/blob/master/exercises/simple-cipher/description.md), the random key part is actually the third and last step in the description. To me, this implies that the test cases involving the random key should also be moved to the bottom of the test cases.

Reply:
The third step is making it fault tolerant. The first step is the cipher without key assignment, that's the random key. But yeah, I will organize the order.

Reply:
@gustavosobral Is is correct that you have not yet applied the re-ordering?

Reply:
The first step is the Cipher without key, so that's the `Random key` set of cases. The second is step is the cypher accepting keys, this is the `Substitution` set of cases. The third step is making it fault tolerant, that's the `Incorrect key` cases. I believe they are in order now, no?

Reply:
Aha, I think I did not explain it correctly. What I meant was not to re-order the individual test cases with the "Random key cipher" case, but the "Random key cipher" case itself. At the moment, the order is:

1. "Random key cipher"
1. "Substitution cipher"
1. "Incorrect key cipher"

I would argue that the first case should actually be the last, as it is a special case of the "Substitution cipher" case. Furthermore, the README also lists it as the third case.

Reply:
Oh, I see what you mean. So if I understood correctly, your suggestion is to change the order to:

1. Substitution cipher
2. Random key cipher
3. Incorrect key cipher

Right? I think we all agree that the last step is making it fault tolerant against incorrect keys?

I see your point and believe that we can really see the Random key as a expansion of the Substitution part, but the thing is that in the exercise [description.md](https://github.com/exercism/problem-specifications/blob/master/exercises/simple-cipher/description.md) it explicitly say that the step 2 is the random key cipher:

> ### Step 2
> Shift ciphers are no fun though when your kid sister figures it out. Try amending the code to allow us to specify a key and use that for the shift distance. This is called a substitution cipher.

So, the exercise description sees the functionality of accepting a key parameter as a expansion of having the random generated case. Which I also believe is correct. Did you see the point here?

Would you still suggest the re-order in this case?

Thank you so much for the feedback and sorry for the late reply.

Reply:
Ah, I then misread the description. It's fine to leave it as-is then.